### PR TITLE
Fixup importing built-ins from module "subdirectories"

### PR DESCRIPTION
**Pull request by jasnell**
there's probably more elegant additional cleanups that can be made in here but this fixes the immediate issue.

### CODE REVIEW

File: src/workerd/jsg/modules.c++ [hunk 1]
```diff
@@ -587,7 +605,16 @@ v8::Local<v8::Value> NodeJsModuleContext::require(jsg::Lock& js, kj::String spec
   auto modulesForResolveCallback = jsg::getModulesForResolveCallback(js.v8Isolate);
   KJ_REQUIRE(modulesForResolveCallback != nullptr, "didn't expect resolveCallback() now");
 
-  kj::Path targetPath = path.parent().eval(specifier);
+  kj::Path targetPath = ([&] {
+    // If the specifier begins with one of our known prefixes, let's not resolve
+    // it against the referrer.
+    if (specifier.startsWith("node:") ||
+        specifier.startsWith("cloudflare:") ||
+        specifier.startsWith("workerd:")) {
```
Comment (dom96) [inline src/workerd/jsg/modules.c++:613]: Non-blocking: would be nice to refactor this out into a separate function to avoid duplication.

**Review by dom96 — APPROVED** [review]
Nice!

File: src/workerd/jsg/modules.c++ (hunk 1 above)
Comment (jasnell) [reply src/workerd/jsg/modules.c++:613]: As a follow on to the module registry refactor I expect much of this to be cleaned up anyway so I'm not that concerned about the duplication right now.

File: src/workerd/api/tests/module-test.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import * as assert from 'a/b/c';
+import * as assert3 from 'node:assert';
+
+export const abortcontroller = {
```
Comment (jasnell) [inline src/workerd/api/tests/module-test.js:4]:
```suggestion
export const basics = {
```

**Pull request merged**